lib/http/request/index.js

```diff
@@ -33,6 +34,9 @@ module.exports = function request(options) {
     options.agent.maxSockets = _MAX_SOCKETS;
   }
 
+  var protocol = 'http' + (options.secure ? 's' : '') + '://';
```

Comment:
There is probably some way we can combine this line 37 w/ line 28 (especially as `options.protocol` doesn't appear to be used anymore... legacy artifact I guess), but oh well, it can/should be handled at another point.